packages/web3-react/src/hooks/useWallet.tsx

```diff
@@ -71,3 +72,27 @@ export function useWallet() {
         }
   }, [signerProvider, account, network, connectionStatus])
 }
+
+export type WalletConfig = {
+  network: NetworkId
+  setNetwork: (network: NetworkId) => void
+  addressGroup?: number
+  setAddressGroup: (addressGroup: number | undefined) => void
+  keyType: KeyType
+  setKeyType: (keyType: KeyType) => void
+}
+
+export function useWalletConfig() {
+  const { network, setNetwork, addressGroup, setAddressGroup, keyType, setKeyType } = useAlephiumConnectContext()
```

Comment:
Question: We disconnect when the `network` changes for now. Do we also want to disconnect when the `addressGroup` and `keyType` change?

Reply:
Great catch. Just added the fix.